src/agent/onefuzz-supervisor/src/coordinator.rs

```diff
@@ -54,21 +54,49 @@ pub struct NodeEventEnvelope {
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
-#[serde(rename_all = "snake_case", untagged)]
+#[serde(rename_all = "snake_case")]
 pub enum NodeEvent {
-    StateUpdate { state: NodeState },
-    WorkerEvent { event: WorkerEvent },
+    StateUpdate(StateUpdateEvent),
+    WorkerEvent(WorkerEvent),
 }
 
-impl From<NodeState> for NodeEvent {
-    fn from(state: NodeState) -> Self {
-        NodeEvent::StateUpdate { state }
+impl From<WorkerEvent> for NodeEvent {
+    fn from(event: WorkerEvent) -> Self {
+        NodeEvent::WorkerEvent(event)
     }
 }
 
-impl From<WorkerEvent> for NodeEvent {
-    fn from(event: WorkerEvent) -> Self {
-        NodeEvent::WorkerEvent { event }
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case", tag = "state")]
+pub enum StateUpdateEvent {
```

Comment:
This may look a little redundant, but in an upcoming PR, we will add additional data to some of these events.